### PR TITLE
Enable to configure allowed HTTP methods for subscriptions

### PR DESCRIPTION
**Pull request by dan-lee**
This is a follow-up PR for #224

Not sure about the wordy option name `allowedSubscriptionHttpMethods`; open for suggestions ✌️

### CODE REVIEW

**Review by n1ru4l — APPROVED** [review]

File: packages/core/lib/process-request.ts [hunk 1]
```diff
@@ -97,6 +97,7 @@ export const processRequest = async <TContext = {}, TRootValue = {}>(
     validate = defaultValidate,
     validationRules,
     variables,
+    allowedSubscriptionHttpMethods = ["GET"],
```
Comment (n1ru4l) [inline packages/core/lib/process-request.ts:100]:
```suggestion
    allowedSubscriptionHttpMethods = ["GET", "POST"],
```

Comment (n1ru4l) [reply packages/core/lib/process-request.ts:100]: We should also allow POST be default even though sse is usually not done via POST.

Comment (n1ru4l) [reply packages/core/lib/process-request.ts:100]: @dan-lee After some more discussions I think we should allow POST by default, can you please do this change and then rebase?

Comment (dan-lee) [reply packages/core/lib/process-request.ts]: @n1ru4l Alright, I applied the change 👍

**Pull request merged**